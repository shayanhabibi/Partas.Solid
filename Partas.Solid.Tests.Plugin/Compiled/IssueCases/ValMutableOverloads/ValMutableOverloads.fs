module Partas.Solid.Tests.IssueCases.ValMutableOverloads

open Partas.Solid
open Fable.Core

[<Erase>]
type MemberVal() =
    interface RegularNode
    [<Erase>]
    member val index: int = unbox null with get,set 
    member this.overloaded
        with inline set(value: string) = this.index <- unbox value

[<Erase>]
type ValMutable() =
    interface RegularNode
    [<DefaultValue>]
    val mutable index: int
    member this.overloaded
        with inline set(value: string) = this.index <- unbox value

[<SolidComponent>]
let TestCase() =
    Fragment() {
        MemberVal(overloaded = "test")
        ValMutable(overloaded = "test")
    }
