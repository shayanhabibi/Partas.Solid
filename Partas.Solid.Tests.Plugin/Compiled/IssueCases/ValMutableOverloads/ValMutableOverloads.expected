export function TestCase() {
    return <>
        <MemberVal index="test" />
        <ValMutable index="test" />
    </>;
}